### DISCUSSION TITLE
Introducing a new core worflow based on the League of Legends API

### DISCUSSION THREAD

**Discussion by TheRustifyer** [zerodaycode/Rumble-LoL-Plugin#7] [Ideas]
I am starting the discussion to decide the best way to implement the new core workflow, when instead of using `OpenCV`
to parse image and act accordingly (based on the voice input) we can directly make REST requests to the League of Legends Client or connect via WebSocket to listen produced events

**Comment by TheRustifyer** [comment]
@Pylyv @Gbm25 @0nSystem well use discusions to decide the best way to introduce the changes exposed above :)

**Comment by gbm25** [comment]
I'm going to copy the message I sent by Discord the other day, so we have all the information here and we can add what we discover.

**Setting Up for Work:**
1. **Postman**: Request access to the workspace, download [this zip](https://github.com/zerodaycode/Rumble-LoL-Plugin/files/13492144/LCU.zip), in it you will find the RIOT certificate, that you will need to install in Postman (Guide [here](https://learning.postman.com/docs/sending-requests/certificates/) ), and an environment that you must import, with the variables that we use.
2. Ensure the client is open and logged in.

**Retrieving Client Connection Data:**
- Use the following command: `type "Path\DelLol\lockfile"`
- This should output in the format: `LeagueClient:12345:54321:password:https`
- The output is divided by ':'. The third item is the port, and the fourth is the password.

**Updating Environment Variables:**
- Update the 'port' and 'pass' variables in the environment to make it ready for use.

For more information and alternative methods, visit: [HextechDocs - Getting Started with the LCU API](https://web.archive.org/web/20230220004724/https://hextechdocs.dev/getting-started-with-the-lcu-api/)

**Relevant Requests:**
1. **To Retrieve Game Modes/Lobbies:**
   - Method: GET
   - Endpoint: `/lol-game-queues/v1/queues`

2. **To Change Lobbies:**
   - Method: POST
   - Endpoint: `/lol-lobby/v2/lobby`
   - Body*: `{"queueId": }` (For ARAM, use `{"queueId": 450}`. For custom matches, it's more complex.)

3. **To Choose Position:**
   - Method: PUT
   - Endpoint: `/lol-lobby/v1/lobby/members/localMember/position-preferences`
   - Body: `{"firstPreference": "String", "secondPreference": "String"}`

4. **To Search for a Match:**
   - Method: POST
   - Endpoint: `/lol-lobby/v2/lobby/matchmaking/search`

5. **To Accept a Match:**
   - Method: POST
   - Endpoint: `/lol-matchmaking/v1/ready-check/accept`

**Useful Links:**
- LCU Swagger-like interface: [LCU and RiotClient API](https://github.com/KebsCS/lcu-and-riotclient-api)
- C++ program for LCU: [KBotExt](https://github.com/KebsCS/KBotExt) (See GameTab.h)
- Unofficial Docs for Riot's features: [Riot API Libraries](https://riot-api-libraries.readthedocs.io/en/latest/lcu.html#tips)
- LCU WebSocket (with Rust listener): [Getting Started with the LCU WebSocket](https://hextechdocs.dev/getting-started-with-the-lcu-websocket/)
- Online version of DataDragon: [CommunityDragon](https://www.communitydragon.org/)
- Many links for Riot-related stuff: [Awesome League by CommunityDragon](https://github.com/communitydragon/awesome-league)
